### PR TITLE
Refactor constraint type meta inf to wrapper

### LINKED ISSUE #435: [BUG] No repeated execution with interpreter-lib
## Steps to reproduce
1. use `interpretString` from `interpreter-lib` twice

For example, in `apps/interpreter/src/examples-smoke-test.spec.ts`

## Description
- Expected: it executes the model twice
- Actual: assertion error on some constraint being registered twice

## Dirty fix in clients using interpreter-lib
```
// Clear registries
metaInformationRegistry.clear();
blockExecutorRegistry.clear();
constraintExecutorRegistry.clear();
```

### PR DESCRIPTION
**Pull request by georg-schwarz**
Similar to #445, but with constraint types.
Should close #435 (to be tested).

- Removes the `ConstraintMetaInfoRegistry` and reads the `ConstraintTypes` from `jv` files instead.
- `ConstraintMetaInformation` becomes a wrapper (static `canBeWrapped` method and parsing via constructor)
- Adds utility function to read all constraint types from std lib instead
- use `await initializeWorkspace(services);` in many tests to make sure the constraint types are loaded properly
-  reorganize tests

Next up:
- Refactor naming of meta inf classes (are actually wrappers now). Deliberately not part of this PR to keep changes small (#460).
- Experiment moving constraint executors to the language server to enable usage of constraints and value types to define block types

### CODE REVIEW

**Comment by georg-schwarz** [comment]
One thing I noticed is the inconsistency in naming (will be addressed in a follow-up PR). We called the block types `blocktype`. For constraint types, `constrainttype` looks less kinda awkward, so I used camel case: `constraintType` in most places (except the grammar). Would love to here your thoughts how I should refactor it. Camel case everywhere (except the grammar)?

**Comment by rhazn** [comment]
> One thing I noticed is the inconsistency in naming (will be addressed in a follow-up PR). We called the block types `blocktype`. For constraint types, `constrainttype` looks less kinda awkward, so I used camel case: `constraintType` in most places (except the grammar). Would love to here your thoughts how I should refactor it. Camel case everywhere (except the grammar)?

CamelCase everywhere has my vote.

File: libs/language-server/src/stdlib/builtin-constrainttypes/RangeConstraint.jv
Comment (rhazn) [inline libs/language-server/src/stdlib/builtin-constrainttypes/RangeConstraint.jv:24]: An inline comment for why we chose this magic value would be nice.

Comment (rhazn) [reply libs/language-server/src/stdlib/builtin-constrainttypes/RangeConstraint.jv:24]: Same on all magic max/min values.

File: libs/language-server/src/stdlib/builtin-constrainttypes/RegexConstraint.jv
Comment (rhazn) [inline libs/language-server/src/stdlib/builtin-constrainttypes/RegexConstraint.jv:8]: I think we were told in MADe that the format is wrong for IFOPT actually? Maybe can just use a different example like an IPv4 address.

File: libs/language-server/src/stdlib/builtin-constrainttypes/LengthConstraint.jv
Comment (rhazn) [inline libs/language-server/src/stdlib/builtin-constrainttypes/LengthConstraint.jv:8]: Including 0 and the max? Or not including?

Comment (rhazn) [reply libs/language-server/src/stdlib/builtin-constrainttypes/LengthConstraint.jv:8]: Also, why this weird number in an example. Why not 1-10 for the VIN10 number we have in another jayvee example?

**Review by rhazn — APPROVED** [review]
Left some small comments, mostly on examples but doesn't need a second review, I trust you choice on those ;).

File: libs/language-server/src/lib/ast/model-util.ts
Comment (georg-schwarz) [reply libs/language-server/src/lib/ast/model-util.ts:265]: IMO not possible as we alter `visitedBuiltinConstraintTypeDefinitions` in the method. Filtering before would lead to a different behavior.

File: libs/language-server/src/lib/ast/model-util.ts
Comment (georg-schwarz) [reply libs/language-server/src/lib/ast/model-util.ts:269]: Possible to do here, but then we would split the pattern:
```
if(Wrapper.canBeWrapped(obj)) {
  const wrapper = new Wrapper(obj);
}
```

I'd rather have this cohesive here to have clear comprehension that the pattern was used correctly.

File: libs/language-server/src/lib/validation/checks/constrainttype-specific/property-assignment.ts
Comment (georg-schwarz) [reply libs/language-server/src/lib/validation/checks/constrainttype-specific/property-assignment.ts:14]: What do you mean by "repeated"?
`property-assigment.ts` deals with validations of single property assignments.
`property-body.ts` deals with validations of a completely property body which means between property assignments.

We might join both files and have two different top-level `check` functions. 
But that breaks the pattern we use for validations: one file per grammatical entity.
So I'd rather keep it like that.

Btw, the `property-assignment` files will vanish as soon as we move the constraint executors to the language-server. Then we don't need those anymore (if we can express them with value types).

Comment (rhazn) [reply libs/language-server/src/lib/validation/checks/constrainttype-specific/property-assignment.ts:14]: Ah yeah that makes sense. I totally missed that checking min > max is only possible if you know both values of course. My bad and this makes sense as it is.

File: libs/language-server/src/lib/ast/model-util.ts
Comment (rhazn) [reply libs/language-server/src/lib/ast/model-util.ts:265]: Fair enough, these are also all just minor code style comments so I am happy with you choosing to not use the filter approach for either as well.

File: libs/language-server/src/lib/validation/checks/constrainttype-specific/property-assignment.ts
Comment (georg-schwarz) [reply libs/language-server/src/lib/validation/checks/constrainttype-specific/property-assignment.ts:14]: I'm also not too happy with that implicit knowledge here to understand it. But I think it will be solved by self-describing the constraints via valuetypes.

**Pull request merged**
